lib/matplotlib/animation.py

```diff
@@ -826,10 +824,10 @@ def setup(self, fig, outfile, dpi, frame_dir=None):
 
         if not self.embed_frames:
             if frame_dir is None:
-                frame_dir = root + '_frames'
-            if not os.path.exists(frame_dir):
-                os.makedirs(frame_dir)
-            frame_prefix = os.path.join(frame_dir, 'frame')
+                frame_dir = Path(str(outfile.with_suffix('')) + '_frames')
+            if not frame_dir.exists():
+                frame_dir.mkdir(parents=True)
+            frame_prefix = str(frame_dir / 'frame')
```

Comment:
Why coerce to string here?

Reply:
I was worried the super() method would not handle Path objects, but it looks like it does.